src/main/java/Duke.java

```diff
@@ -1,10 +1,260 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.nio.file.Paths;
+
 public class Duke {
+    private static final String logo = " ____        _        \n"
+                                + "|  _ \\ _   _| | _____ \n"
+                                + "| | | | | | | |/ / _ \\\n"
+                                + "| |_| | |_| |   <  __/\n"
+                                + "|____/ \\__,_|_|\\_\\___|\n";
+
+    private static final String LINE = "\t____________________________________________________________";
+
+    private Scanner sc = new Scanner(System.in);
+    private ArrayList<Task> list = new ArrayList<>();
+
+    private static enum Command {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE,
+    }
+
+    private static final String filePath = Paths.get("data", "duke.txt").toString();
+    private File dataFile = new File(filePath);
+
+    private void sayGreetings() {
+        System.out.println(LINE);
+        System.out.println("\tHello! I'm SKY");
+        System.out.println("\tWhat can I do for you?");
+        System.out.println(LINE);
+    }
+
+    private void sayBye() {
+        System.out.println("\tBye. Hope to see you again soon!");
+        System.out.println(LINE);
+    }
+
+    private void echo(String input) {
+        System.out.println(input);
+    }
+
+    private void add(Task task) {
+        this.list.add(task);
+        System.out.println("\tGot it. I've added this task:");
+        System.out.println("\t  " + task);
+        System.out.println("\tNow you have " + this.list.size() + " tasks in the list.");
+    }
+
+    private void list() {
+        System.out.println("\tHere are the tasks in your list:");
+        for (int i = 0; i < this.list.size(); i++) {
+            System.out.println("\t" + (i + 1) + ". " + this.list.get(i));
+        }
+    }
+
+    private void mark(String[] input) throws IncompleteCommandException, InvalidArgumentException {
+        if (input.length == 1 || input[1].equals("")) {
+            throw new IncompleteCommandException(input[0]);
+        } 
+        try {
+            int index = Integer.parseInt(input[1]) - 1;
+            this.list.get(index).markAsDone();
+        } catch (IndexOutOfBoundsException e) {
+            throw new InvalidArgumentException();
+        }
+    }
+
+    private void unmark(String[] input) throws IncompleteCommandException, InvalidArgumentException {
+        if (input.length == 1 || input[1].equals("")) {
+            throw new IncompleteCommandException(input[0]);
+        } 
+        try {
+            int index = Integer.parseInt(input[1]) - 1;
+            this.list.get(index).markNotDone();
+        } catch (IndexOutOfBoundsException e) {
+            throw new InvalidArgumentException();
+        }
+    }
```

Comment:
Method names could be more descriptive, for example 
`private void addTask(Task task)`
`private void listAllTasks()`